### DISCUSSION TITLE
datasette requires asgi requires unix?

### DISCUSSION THREAD

**Discussion by jchalfant** [simonw/datasette#1093] [Q&A]
Hi All,

Very glad for this project and eager to begin using on Windows. But I can't get datasette properly installed,..
- fully patched Win10 (Windows 10 Education N, 20H2, 19042.572)
- fresh install of Python 3.9 (.msi from Python.org)
- fresh install of datasette (pip install datasette)

I run datasette and:
Traceback (most recent call last):
  File "c:\program files\python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
   ...
  File "c:\program files\python39\lib\site-packages\datasette\utils\asgi.py", line 2, in <module>
    from os import EX_CANTCREAT
ImportError: cannot import name 'EX_CANTCREAT' from 'os' (c:\program files\python39\lib\os.py)

Sure enough, my ~\python39\lib\os.py does not contain 'EX_CANTCREAT'. 

Then I research https://docs.python.org/3.9/library/os.html and see:
os.EX_CANTCREAT
Exit code that means a user specified output file could not be created.
**Availability: Unix.**

So apparently we are not expected to find EX_CANTCREAT on Windows. But are we also not expected to be able to run datasette on Windows?

Or did I miss something? (Or several things?)

Any help, especially pointers to docs, will be appreciated.

Thank you!

-J

**Comment by simonw** [comment]
I would really like to get Datasette running on Windows, but I've not put in the work yet. I need:

- to figure out my own Windows Python development environment (I have a gaming machine I can use for this)
- to get GitHub Actions to run Datasette CI tests on Windows
